src/nbl/video/IGPUDescriptorSet.cpp

```diff
@@ -30,38 +30,9 @@ IGPUDescriptorSet::~IGPUDescriptorSet()
 {
     if (!isZombie())
     {
-        const bool allowsFreeing = m_pool->m_creationParameters.flags.hasFlags(IDescriptorPool::ECF_FREE_DESCRIPTOR_SET_BIT);
-        for (auto i = 0u; i < static_cast<uint32_t>(asset::IDescriptor::E_TYPE::ET_COUNT); ++i)
-        {
-            // There is no descriptor of such type in the set.
-            if (m_descriptorStorageOffsets.data[i] == ~0u)
-                continue;
-
-            const auto type = static_cast<asset::IDescriptor::E_TYPE>(i);
-
-            const uint32_t allocatedOffset = getDescriptorStorageOffset(type);
-            assert(allocatedOffset != ~0u);
-
-            const uint32_t count = m_layout->getTotalDescriptorCount(type);
-            assert(count != 0u);
-
-            std::destroy_n(m_pool->getDescriptorStorage(type) + allocatedOffset, count);
-
-            if (allowsFreeing)
-                m_pool->m_descriptorAllocators[i]->free(allocatedOffset, count);
-        }
-
-        const auto mutableSamplerCount = m_layout->getTotalMutableSamplerCount();
-        if (mutableSamplerCount > 0)
-        {
-            const uint32_t allocatedOffset = getMutableSamplerStorageOffset();
-            assert(allocatedOffset != ~0u);
-
-            std::destroy_n(m_pool->getMutableSamplerStorage() + allocatedOffset, mutableSamplerCount);
-
-            if (allowsFreeing)
-                m_pool->m_descriptorAllocators[static_cast<uint32_t>(asset::IDescriptor::E_TYPE::ET_COUNT)]->free(allocatedOffset, mutableSamplerCount);
-        }
+        m_pool->deleteSetStorage(this);
+        if (m_pool->m_creationParameters.flags.hasFlags(IDescriptorPool::ECF_FREE_DESCRIPTOR_SET_BIT))
+            m_pool->m_descriptorSetAllocator.free_addr(m_poolOffset, 1);
```

Comment:
I'd roll the `m_descriptorSetAllocator` stuff into the `deleteSetStorage`

Reply:
> Resolved, but I don't get what you mean by:
> 
> > (furthermore if it doesn't allow freeing you could check if the num

basically inside the `deleteSetStorage` if you see that there are no more non-zombie descriptor sets, you can call `reset()` implcitly

Reply:
Okay, don't really know what the use case of this is but.. resolved, I guess.